programs/fileio.c

```diff
@@ -769,20 +769,24 @@ static size_t FIO_createDictBufferMMap(void** bufferPtr, const char* fileName, F
     close(fileHandle);
     return (size_t)fileSize;
 }
-static void FIO_munmapDictBuffer(void* dictBuffer, size_t dictBufferSize) {
-    FIO_munmap(dictBuffer, dictBufferSize);
-}
 #else
 static size_t FIO_createDictBufferMMap(void** bufferPtr, const char* fileName, FIO_prefs_t* const prefs, stat_t* dictFileStat)
 {
    return FIO_createDictBuffer(bufferPtr, fileName, prefs, dictFileStat);
 }
-static void FIO_munmapDictBuffer(void* dictBuffer, size_t dictBufferSize) {
-   (void)dictBufferSize;
-   free(dictBuffer);
+static void FIO_munmap(void* buffer, size_t bufferSize) {
+   (void)bufferSize;
+   free(buffer);
 }
 #endif
 
+static void FIO_freeDict(const FIO_Dict_t* dict) {
+    if (dict->dictBufferType == FIO_mallocDict) {
+        free(dict->dictBuffer);
+    } else if (dict->dictBufferType == FIO_mmapDict)  {
+        FIO_munmap(dict->dictBuffer, dict->dictBufferSize);
+    }
+}
```

Comment:
minor: I would add an `else assert(0);` at the end, 
because such a case is supposed to never happen,
unless the `FIO_Dict_t` code is changed in the future.